src/main/java/ch/njol/skript/effects/base/SetEffect.java

```diff
@@ -0,0 +1,123 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+/**
+ * Contains the default effects of Skript.
+ * 
+ * @author Peter Güttinger
+ */
+package ch.njol.skript.effects.base;
+
+import java.util.Arrays;
+import java.util.function.BiConsumer;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptAPIException;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+/**
+ * A class that is to be used for boolean expressions.
+ * <p>
+ * A boolean expression should be a condition for getting it's value.
+ */
+public abstract class SetEffect<T> extends Effect {
+
+	private Expression<Boolean> value;
+	private Expression<T> expression;
+	private boolean make;
+
+	/**
+	 * Registers an effect with patterns "set property of %type% to %boolean%" and "set %types%'[s] property to %boolean%"
+	 * 
+	 * @param effect The SetEffect class that you are registering.
+	 * @param property The property of the syntax.
+	 * @param type The type classinfo of the syntax. Can be plural.
+	 */
+	public static void register(Class<? extends SetEffect<?>> effect, String property, String type) {
+		Skript.registerEffect(effect, "set " + property + " of %" + type + "% to %boolean%",
+				"set %" + type + "%'[s] " + property + " to %boolean%",
+				"make %" + type + "% " + property);
+	}
```

Comment:
My concern here is that the `set` and `make` property usually differ. For example:
```
set player's flight to true
make player fly
```
I don't think it'd be good to say `set player's fly to true` or `make player flight`

Reply:
How would you differ that here? How do you think that be resolved here?

Reply:
Resolved